pkg/ssh/connection_golang.go

```diff
@@ -289,11 +289,18 @@ func ValidateAndConfigure(uri *url.URL, iden string, insecureIsMachineConnection
 			keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
 			known, err := knownhosts.New(keyFilePath)
 			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					logrus.Warn("please create a known_hosts file. The next time this host is connected to, podman will add it to known_hosts")
-					return nil
+				if !errors.Is(err, os.ErrNotExist) {
+					return err
+				}
+				k, err := os.OpenFile(keyFilePath, os.O_RDWR|os.O_CREATE, 0o600)
```

Comment:
To be specific about the next steps, I think this should be preceded by some variant of `[ -d ~/.ssh ] || mkdir -m 0700 ~/.ssh ]`.

That way we treat “a `known_hosts` file exists but is empty”, “`.ssh` exists but not `known_hosts`”, and “`.ssh` does not exist” exactly the same.